src/main/java/graphql/nadel/engine/HydrationInputResolver.java

```diff
@@ -275,14 +281,17 @@ private ExecutionResultNode convertSingleHydrationResultIntoOverallResult(Execut
                                                                               HydrationTransformation hydrationTransformation,
                                                                               RootExecutionResultNode rootResultNode,
                                                                               QueryTransformationResult queryTransformationResult,
-                                                                              NadelContext nadelContext) {
+                                                                              NadelContext nadelContext,
+                                                                              ResultComplexityAggregator resultComplexityAggregator) {
 
         synthesizeHydratedParentIfNeeded(fieldTracking, hydratedFieldStepInfo);
 
         Map<String, FieldTransformation> transformationByResultField = queryTransformationResult.getTransformationByResultField();
         Map<String, String> typeRenameMappings = queryTransformationResult.getTypeRenameMappings();
         ExecutionResultNode firstTopLevelResultNode = serviceResultNodesToOverallResult
                 .convertChildren(executionId, forkJoinPool, rootResultNode.getChildren().get(0), overallSchema, hydratedFieldStepInfo, true, false, transformationByResultField, typeRenameMappings, nadelContext);
+        String serviceName = hydrationTransformation.getUnderlyingServiceHydration().getServiceName();
+        resultComplexityAggregator.addAndSetServiceNodeCount(serviceName, firstTopLevelResultNode.getResultNodeCount());
```

Comment:
naming

`addAndSetServiceNodeCount` is not the bets name (you can tell its AtomicInteger) inspired.

just called it `incrementServiceNodeCount` say